BitFaster.Caching/Lfu/NodePolicy.cs

```diff
@@ -1,22 +1,134 @@
     }
 
     internal struct AccessOrderPolicy<K, V> : INodePolicy<K, V, AccessOrderNode<K, V>>
+        where K : notnull
     {
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public AccessOrderNode<K, V> Create(K key, V value)
         {
             return new AccessOrderNode<K, V>(key, value);
         }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public bool IsExpired(AccessOrderNode<K, V> node)
+        {
+            return false;
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public void AdvanceTime()
+        {
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public void OnRead(AccessOrderNode<K, V> node)
+        {
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public void OnWrite(AccessOrderNode<K, V> node)
+        {
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public void OnEvict(AccessOrderNode<K, V> node)
+        {
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public void ExpireEntries<P>(ref ConcurrentLfuCore<K, V, AccessOrderNode<K, V>, P> cache) where P : struct, INodePolicy<K, V, AccessOrderNode<K, V>>
+        {
+        }
+    }
+
+    internal struct ExpireAfterPolicy<K, V> : INodePolicy<K, V, TimeOrderNode<K, V>>
+        where K : notnull
+    {
+        private readonly IExpiryCalculator<K, V> expiryCalculator;
+        private readonly TimerWheel<K, V> wheel;
+        private Duration current;
+
+        public ExpireAfterPolicy(IExpiryCalculator<K, V> expiryCalculator)
+        {
+            this.wheel = new TimerWheel<K, V>();
+            this.expiryCalculator = expiryCalculator;
+            this.current = Duration.SinceEpoch();
+            this.wheel.time = current.raw;
+        }
+
+        public IExpiryCalculator<K, V> ExpiryCalculator => expiryCalculator;
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public TimeOrderNode<K, V> Create(K key, V value)
+        {
+            var expiry = expiryCalculator.GetExpireAfterCreate(key, value);
+            return new TimeOrderNode<K, V>(key, value) { TimeToExpire = Duration.SinceEpoch() + expiry };
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public bool IsExpired(TimeOrderNode<K, V> node)
+        {
+            return node.TimeToExpire < Duration.SinceEpoch();
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public void AdvanceTime()
+        {
+            current = Duration.SinceEpoch();
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public void OnRead(TimeOrderNode<K, V> node)
+        {
+            var currentExpiry = node.TimeToExpire - current;
+            node.TimeToExpire = current + expiryCalculator.GetExpireAfterRead(node.Key, node.Value, currentExpiry);
+            wheel.Reschedule(node);
```

Comment:
Need a guard here to avoid rescheduling the same node into the same bucket if it was read twice or more.